lib/jobs/obm-control.js

```diff
@@ -64,18 +64,33 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
+            //var settings = _.find(node.obmSettings, function(setting) {
+            self.settings = _.find(node.obmSettings, function(setting) {
                 return setting ? setting.service === self.obmServiceName : false;
             });
 
-            assert.ok(settings, "Node should have OBM settings for service: " +
+            if(!self.settings){
```

Comment:
I think it will be more reasonable to find the default obm-setting only if no explict obmServiceName is specified or set to `"auto"`.  Because some code logic actually don't want the default behavior.

The logic looks like:
```javascript
if (self.obmServiceName) {
   // throw error if the exact obm service doesn't exist
} else {
   //find the default obm service
}
```